### ISSUE TITLE
[Due for payment 2025-09-12] [Live Markdown] Support inline code background style

### ISSUE THREAD

**Issue by thienlnam** [Expensify/App#39518]
https://expensify.slack.com/archives/C049HHMV9SM/p1712094742944359

Now that live markdown preview is out on production - let's add support for the background style for inline code

It should use the same font and background style, but currently doesn't.

![image](https://github.com/Expensify/App/assets/30609178/e0e111a5-0d02-4526-8b67-7cea71e1e7dc)

<details><summary>Issue Owner</summary>Current Issue Owner: @lschurr</details>

**Comment by thienlnam** [comment]
Added to the main tracking issue as part of HIGH: https://github.com/Expensify/App/issues/36071

**Comment by lschurr** [comment]
Anything needed from BZ on this one @thienlnam?

**Comment by thienlnam** [comment]
Not at this time, we'll probably need payment to the C+ that reviews in the future but that's further away

**Comment by tomekzaw** [comment]
We're working on it.

### iOS

I've asked @maksg from SWM to implement the iOS part. Here's the draft PR to Live Markdown:
* https://github.com/Expensify/react-native-live-markdown/pull/306

### Android

We'll start working on it once we complete the iOS part.

### Web

@Skalakid had a working PoC some time ago, I will ask him to submit a PR for web.

**Comment by lschurr** [comment]
Any update @thienlnam @tomekzaw?

**Comment by Skalakid** [comment]
Hi @lschurr, @BartoszGrajdek is working on it

**Comment by BartoszGrajdek** [comment]
Hi @lschurr so here's a quick update on what's happening:

There are 2 different blocks to handle here contrary to what the issue title/description may say namely: inline code & pre blocks (single / triple backticks). 

Like we agreed [here](https://swmansion.slack.com/archives/C06BDSWLDPB/p1713369992524139?thread_ts=1713274778.358479&cid=C06BDSWLDPB) there will be some changes to ExpensiMark to limit the number of cases where the `pre` styling is applied - I have it roughly done, but we will likely need some additional refactors.

Currently, I'm working on implementing the `pre` block styles for web. That's a tricky case because of the limitations that the Live Markdown puts on us, but I'm trying to overcome it.

I'll keep you updated! 😄

**Comment by quinthar** [comment]
Hi, what's the next step and ETA on this?

**Comment by tomekzaw** [comment]
Here's the Android PR from @maksg:
* https://github.com/Expensify/react-native-live-markdown/pull/338

Currently waiting for @BartoszGrajdek's PR for web

Next step: Complete PR for web

ETA: ~2 weeks

**Comment by lschurr** [comment]
Any update @BartoszGrajdek @tomekzaw?

**Comment by BartoszGrajdek** [comment]
I've just added a PR to the `expensify-common` with some changes to how we treat code blocks. I'm currently solving the remaining bugs on Live Markdown for web to get it working on all browsers - after that we can move forward with the process of getting implementations for all platforms merged 😄  @lschurr

**Comment by dangrous** [comment]
PR has been merged (accidentally, by me, due to the merge freeze whoops)! So soon this will be good to go but don't merge any package.json changes yet! Also adding myself here for any further assistance since I think the issue linking got weird.

**Comment by lschurr** [comment]
Any update @BartoszGrajdek @dangrous?

**Comment by BartoszGrajdek** [comment]
Hi, it took us more time than we anticipated, mainly due to some bugs we have found that turned out to block us here. We're hoping to close this one in the coming days. There is [one PR](https://github.com/Expensify/expensify-common/pull/705) that will resolve an issue in the ExpensiMark itself which came to our attention after some extensive testing (I'll tag @dangrous to review it once it's ready). 🙏🏻 

@Skalakid is currently testing it together with [the web PR](https://github.com/Expensify/react-native-live-markdown/pull/341). Once we make sure there are no additional bugs we'll push the `expensify-common` PR & later all the PRs that are waiting in the `react-native-live-markdown` lib. 🤓

Starting today I'll give you daily updates to keep you posted, we're getting really close! 🤞🏻

**Comment by BartoszGrajdek** [comment]
Status for today - Michał has found 2 small bugs during the testing. I'm already onto them hoping to solve both tomorrow 😄

**Comment by BartoszGrajdek** [comment]
### Status 5.06.2024

We've managed to track the root cause for both of the bugs. One is a regression created by another PR, the second one has appeared because of the changes we've made and it's almost fixed now. Since none of them are related to ExpensiMark itself I opened up [the PR](https://github.com/Expensify/expensify-common/pull/705) for review (@dangrous if you have the capacity please go ahead & review it 🙏🏻 ).

**Comment by BartoszGrajdek** [comment]
### Status 6.06.2024
I was debugging quite a few things today. We currently have ~2 other blockers at hand that need to be solved before we can add any additional changes, but I managed to work on regression fixes for a bit.

**Comment by BartoszGrajdek** [comment]
### Status 10.06.2024
Didn't forget about this issue, right now we're solving some of the bugs that came up during the bump of `react-native-live-markdown` so I'm busy with that 👀

**Comment by BartoszGrajdek** [comment]
### Status 11.06.2024-13.06.2024

Giving an update here, since we have discussed this issue internally at length after encountering all of the regressions/bugs that we've been fighting for the past week. We think that the best course of action will be to wait temporarily for a refactor that @Skalakid is working on. This will allow us to implement a much cleaner solution on the web that won't be susceptible to regressions. 

I'll try to regularly come back here and let you know what's going on 🙌🏻

**Comment by thienlnam** [comment]
@BartoszGrajdek Is there a PR we can link to that we're waiting on?

**Comment by BartoszGrajdek** [comment]
Of course, here you go: https://github.com/Expensify/react-native-live-markdown/pull/394 🙌🏻

**Comment by lschurr** [comment]
On hold

**Comment by lschurr** [comment]
Hold

**Comment by thienlnam** [comment]
Held PR still in the works

**Comment by BartoszGrajdek** [comment]
Hi! Adding an update here, since a few things have happened. 👀 

As you know we already have the changes for Android/iOS prepared. At the beginning of this week, I finished a web implementation that I had been working on for the past ~2 weeks and it works really well in our refactored web component structure. That means everything's ready waiting for the refactor in question to get merged.

We're launching (hopefully) the last round of testing for the live markdown refactor PR and our current focus is to get it done ASAP. After that, if no regressions arise we'll work on merging all PRs inline-code / codefence related.

I'm OOO together with @Skalakid till 12.08, so I'll update you once we're back and know the test results. 🙌

**Comment by lschurr** [comment]
Anything new @BartoszGrajdek or are we still on hold?

**Comment by dangrous** [comment]
shush melvin

**Comment by tomekzaw** [comment]
We're working on it in `react-native-live-markdown`:
* https://github.com/Expensify/react-native-live-markdown/pull/456
* https://github.com/Expensify/react-native-live-markdown/pull/338
* https://github.com/Expensify/react-native-live-markdown/pull/306

**Comment by lschurr** [comment]
This looks like it's still on hold

**Comment by BartoszGrajdek** [comment]
@lschurr The [web PR](https://github.com/Expensify/react-native-live-markdown/pull/456) is no longer on hold. We're currently working on some remaining bugs together with @Skalakid, and we'll push it to E/App once it's bug-free. 

I'll keep you updated! 🤞🏻

**Comment by twisterdotcom** [comment]
Relabelling as Lauren is OOO now.

**Comment by lschurr** [comment]
Thanks Zans. Removing you since I'm back. This is still on hold.

**Comment by lschurr** [comment]
Still on hold

**Comment by lschurr** [comment]
Hold

**Comment by lschurr** [comment]
Still on hold

**Comment by dangrous** [comment]
Should we move this to Monthly while it's on hold?

**Comment by BartoszGrajdek** [comment]
[Web implementation PR](https://github.com/Expensify/App/pull/56064) is now in draft, I just need to resolve some small issues and we should be able to give it to the QA team for testing (before merging the PR in `react-native-live-markdown`.

Here's a preview:

![Image](https://github.com/user-attachments/assets/a6542139-de3e-43b0-a782-81564de93ef5)

**Comment by BartoszGrajdek** [comment]
Also like I've mentioned [here](https://github.com/Expensify/App/issues/39518#issuecomment-2444761430) this issue is no longer on HOLD, it's just that we're busy with other issues, so we couldn't focus solely on this.

**Comment by dangrous** [comment]
ah okay great, i've updated the title

**Comment by dangrous** [comment]
Do you have a sense of when you might have bandwidth to get back to this one?

**Comment by BartoszGrajdek** [comment]
Hi! I'm working on this part-time, whenever I'm not busy with other issues. 

Last week I solved some bugs related to line-height breaking the code blocks. Currently, I'm waiting to get an adhoc build to test if everything's working and then we can invite either C+ or some people from the QA team to test this before we merge it into `react-native-live-markdown`.

**Comment by BartoszGrajdek** [comment]
Hi! 
Right now we're going back and forth with @dukenv0307 looking for bugs & fixing them if any are found. Once we can't find any the plan looks like this:

1. Merge both `expensify-common` changes as well as `react-native-live-markdown` changes
2. Update the PR I've linked above to only contain package version bumps instead of build `.tgz` files
3. Ensure everything's working fine and it should be ready to merge 😄

**Comment by dangrous** [comment]
Any updates @BartoszGrajdek ?

**Comment by Skalakid** [comment]
Hello, @BartoszGrajdek is no longer working on Expensify. I took over this task. For now no updates on this, because we have to finish higher priority task (Add to Wallet feature). However, the changes are tested in Expensify and we need to merge it inside the Live Markdown and prepare the new PR for NewDot. I will try to do it this or next week :D

**Comment by dangrous** [comment]
Sounds good, thank you @Skalakid !

**Comment by dangrous** [comment]
looks like we're almost there on that PR, right? Let us know how we can help

**Comment by dangrous** [comment]
How are we looking @Skalakid?

Also - should I assign you to this issue? Thanks!

**Comment by Skalakid** [comment]
Hello, the [PR](https://github.com/Expensify/react-native-live-markdown/pull/456) is finished and got the review under it. I fixed all critical bugs connected to these changes and simplified the implementation by doing a refactor. I will address all the review comments, and then after the merge, I will create the PR with the Live Markdown bump to E/App

**Comment by Skalakid** [comment]
@dangrous can you assign someone to review [this PR in expensify-common](https://github.com/Expensify/expensify-common/pull/865) repo, please?

**Comment by dangrous** [comment]
How are we doing @Skalakid?

**Comment by jmusial** [comment]
OK, so confirming from the msg above:
- Web is done
- @war-in  will handle native platforms implementation in `live-markdown`
- PR will be closed as no longer relevant
